src/base-lazy-component.js

```diff
@@ -49,6 +49,7 @@ export default class BaseLazyComponent extends React.Component {
     if (!this.resolvedData) {
       this.resolvedData = suspensePayload.resolvedData;
       this.resourceLoader = suspensePayload.promise;
+      this.component = this.resolvedData.default || this.resolvedData
```

Comment:
this feels like it's forcing the base class to be aware of the derived class implementation.

I guess using the `setState` flow becomes redundant in the suspense case flow so having something like this makes sense.

Wdyt about having `renderComponent` in the base class accept an optional `child` which can be `this.state.component` by default maybe (or maybe getting rid of the base class knowing anything about this is correct and all dervied class can explicitly specify it) then the derived class can pass `this.resolvedData.default || this.resolvedData`.

Also I guess this is super tricky to add a regression test for?